src/runtime/task/clock.rs

```diff
@@ -0,0 +1,131 @@
+use crate::runtime::task::{TaskId, DEFAULT_INLINE_TASKS};
+use smallvec::{smallvec, SmallVec};
+use std::cmp::{Ordering, PartialOrd};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct VectorClock {
+    pub(crate) time: SmallVec<[u32; DEFAULT_INLINE_TASKS]>,
+}
+
+impl VectorClock {
+    pub(crate) fn new() -> Self {
+        Self { time: SmallVec::new() }
+    }
+
+    #[cfg(test)]
+    pub(crate) fn new_from(v: &[u32]) -> Self {
+        let time: SmallVec<[u32; DEFAULT_INLINE_TASKS]> = SmallVec::from(v);
+        Self { time }
+    }
+
+    // Zero extend clock to accommodate `task_id` tasks.
+    pub(crate) fn extend(&mut self, task_id: TaskId) {
+        let d = 1 + task_id.0 - self.time.len();
```

Comment:
nit: can we pick a better variable name than `d`? :-)

Reply:
How about `d1` ? ;-)

Reply:
Renamed to `num_new_tasks`